homeassistant/components/google_assistant/trait.py

```diff
@@ -535,10 +535,20 @@ def supported(domain, features):
     def sync_attributes(self):
         """Return temperature point and modes attributes for a sync request."""
         modes = []
-        for mode in self.state.attributes.get(climate.ATTR_OPERATION_LIST, []):
-            google_mode = self.hass_to_google.get(mode)
-            if google_mode is not None:
-                modes.append(google_mode)
+        supported = self.state.attributes.get(ATTR_SUPPORTED_FEATURES)
+
+        if supported & climate.SUPPORT_ON_OFF != 0:
```

Comment:
On/off should be picked up by the on/off trait not by the temp mode trait.

Reply:
It should be, but it isn't. GA won't call the on/off trait at all for climate devices. If you don't report this then if the device is off GA reports it as can't connect.

Reply:
If you send 'on' and 'off' more than once in the list it doesn't matter, GA just ignores the duplicate. If you say "Hey google turn \<device\> on" then it sends "on" as the target state, but then reports back whatever state it ends in.

Reply:
> "On" will need to be added like you have but you should test to see what HA does when you send that state.

Makes the most sense to me to call the turn on service. That follows the intent of the command I feel. If the device doesn't have on/off then it doesn't matter since I'm not sending "on/off" here.

Reply:
When you send the on command via GA what does it send us? Does it send on or the previous state?